### ISSUE TITLE
Max 2d position embeddings in layoutlm

### ISSUE THREAD

**Issue by lumliolum** [microsoft/unilm#357]
Hello,
I have been reading layoutlm, layoutlmv2 paper, and reading the codebase to better understand model architecture.

So I have some of the questions which I listed below

1. In both layoutlm and layoutlmv2 paper, the bounding box coordinates (that is given by ocr for each word) are normalized to the range of 0-1000. But when ever I check the code for constructing position_embeddings
```python
# taken from modelling_layoutlmv2.py
self.x_position_embeddings = nn.Embedding(config.max_2d_position_embeddings, config.coordinate_size)
self.y_position_embeddings = nn.Embedding(config.max_2d_position_embeddings, config.coordinate_size)
self.h_position_embeddings = nn.Embedding(config.max_2d_position_embeddings, config.shape_size)
self.w_position_embeddings = nn.Embedding(config.max_2d_position_embeddings, config.shape_size)
self.token_type_embeddings = nn.Embedding(config.type_vocab_size, config.hidden_size)
```
Here I see the value of `config.max_2d_position_embeddings = 1024`. Can I know why this value is 1024 and not 1000 (if one of the coordinates is equal to width or height, then the normalized value will be 1000, so we should take the size 1001 but I don't understand why 1024)?

2. In layoutlm v1, the  embeddings (from the bounding box) are calculated as shown by the code snippet 
```python
# taken from modelling/layoutlm.py
# in forward function
embeddings = (
  words_embeddings
  + position_embeddings
  + left_position_embeddings
  + upper_position_embeddings
  + right_position_embeddings
  + lower_position_embeddings
  + h_position_embeddings
  + w_position_embeddings
  + token_type_embeddings
)
```
and also the dimensions of `left_position_embeddings, upper_position_embeddings, lower_position_embeddings, h_position_embeddings, w_position_embeddings` is `768`. (you can see that all the embeddings are added)

But in layoutlm v2 the embeddings are concatenated (given on page 4 of the paper, section: layoutlm embeddings). Why spatial embeddings are concatenated in v2. Was there any particular reason why they are doing this. Did the authors found the improvement in this approach ?

**Comment by NielsRogge** [comment]
For the first question, see #304

**Comment by lumliolum** [comment]
> For the first question, see #304

Thanks for the link. that was also my first guess but I thought that there may be something hidden in 1024 :stuck_out_tongue:
